### ISSUE TITLE
Join room prompt asks a question with only one possible answer

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#9609]
![image](https://user-images.githubusercontent.com/1190097/57043506-f6fe6380-6c24-11e9-852d-04a5a9ac78f5.png)

There's no button for "no, I don't want to do that" which leaves me wondering what I'm supposed to do. I don't really know what the button would do in this scenario, so would possibly recommend rewording the question to not be a question. Something like "#room can't be previewed - join it to see what's going on".

A possible action for a "no" button could be "go back to what you were doing", although that feels needlessly complicated.

**Comment by lampholder** [comment]
I agree; @nadonomy do you have any thoughts on the wording?

As a related point, could we take this opportunity to reinforce _why_ some rooms can be previewed and not others? It needn't be in large writing - just a small amount of additional 2nd tier copy for the interested - something like:

## __Room Name__ can't be previewed - join it now to see what's going on.

This room is not on your homeserver, and rooms cannot yet be previewed over federation. You will have to join in order to read the messages.

_or_

The room administrators have disabled preview for this room. You will have to join in order to read the messages.

**Comment by ara4n** [comment]
I would strongly suggest keeping the amount of text minimal here. The user doesn’t need to know that peeking is disabled over federation.

When do we show this anyway? If I click on a link to join a room i want to just join it, not have a double opt in

**Comment by ara4n** [comment]
I guess it’s to let users confirm if they were assuming the room has a preview. Perhaps:

Room name
Avatar
Preview not available
[Join button]

?

**Comment by nadonomy** [comment]
> When do we show this anyway? If I click on a link to join a room i want to just join it, not reconfirm

I'm assuming that this isn't shown for invites (please someone feel free to correct me). Considering a user will have arrived by either (a) clicking on a matrix.to link or (b) clicking on a link to a room in their client I'm inclined to agree.

> I guess it’s to let users confirm if they were assuming the room has a preview. Perhaps:
> 
> ```
> <Room name>
> <Avatar>
> “Preview not available”
> [Join button]
> ```
> 
> ?

This looks like a reasonable iteration on the screenshot above, but I'm not sure it actually addresses @turt2live's concern of not having a way to cancel. @turt2live is the issue that the room exists in the room list and you have no way to get rid of it? Or something else?

**Comment by dbkr** [comment]
I'm interpreting the concern as more that it asks you a question but there's only one option button which is a bit jarring. Since it no longer asks you the question, I imagine this is not a problem and you'd just cancel by navigating away.

**Comment by ara4n** [comment]
i think his issue was that it was a question with only one answer. by making it a statement with only one action, the problem goes away. if you tap on a different entry in the roomlist you automatically move on.

**Comment by lampholder** [comment]
> I would strongly suggest keeping the amount of text minimal here. The user doesn’t need to know that peeking is disabled over federation.

Just to explore this a tiny bit more - I suspect very few people understand when peeking would work and when it wouldn't, meaning people don't know where to start if they want to make a room peekable. 

Personally I find it very frustrating when a service tells me what is happening but not why. It means I've been denied an opportunity to learn, and I'm left typing symptoms into Google if I want to address the issue. 

We could even hide the detailed text behind some 'more details' UX perhaps?

**Comment by turt2live** [comment]
My concern is indeed that a question is being asked but there's only one possible answer. Making it a statement would solve the issue.

I also got this prompt from clicking a pill on a room I haven't joined yet - I'd prefer as a user if that tried to preview first before I commit to joining.

**Comment by nadonomy** [comment]
> My concern is indeed that a question is being asked but there's only one possible answer. Making it a statement would solve the issue.

Stripping "Do you want to join it?" sounds good to me in that case, along with adding the room avatar (as we do in some other room preview contexts already) as per Matthews request.

**Comment by nadonomy** [comment]
> Personally I find it very frustrating when a service tells me what is happening but not why. It means I've been denied an opportunity to learn, and I'm left typing symptoms into Google if I want to address the issue.

I agree adding more info can be useful for advanced users. As the very large majority of users aren't as sympathetic to the protocol as we are, nor should they be, it'd be terrible to present users with a wall of text.

> We could even hide the detailed text behind some 'more details' UX perhaps?

In future, I'd love to start using little (?) icons throughout the app which reveal more info in tooltips, and link out to help documentation. Should we track that as a meta issue?

**Comment by lampholder** [comment]
> I agree adding more info can be useful for advanced users. As the very large majority of users aren't as sympathetic to the protocol as we are, nor should they be, it'd be terrible to present users with a wall of text.

This isn't just a protocol issue. Riot/Matrix room permissions are baffling; some of that we can simplify in the client, some of that we can't work around in the client and will have to explain. If we do neither, regardless of one's level of expertise Riot will be an unpredictable (and hence untrustworthy) platform.

Event today I am not confident that I would be able to configure a room to give the exact experience I want users to have, because Riot does a poor job of training me or explaining to me which options have which consequences.

> Should we track that as a meta issue?

Feel free :) Some of the detailed concepts are complex, though - will we be able to fit everything we need to in a tooltip?

**Comment by bkil** [comment]
This is somewhat also related to #21288.

It has also happened to me a few times that I pressed `Preview` on the `Explore Rooms` dialog and after loading for a few seconds, I got this same error message that I _must_ join instead.

**Comment by t3chguy** [comment]
> It has also happened to me a few times that I pressed Preview on the Explore Rooms dialog and after loading for a few seconds, I got this same error message that I must join instead.

This would be an issue with your server not being able to peek into what it claimed is a peekable room, nothing the client can do about that.

**Comment by bkil** [comment]
Sure, I'm not asking to try to fix from the client side, rather to provide a UX for previewing rooms without losing the position in the window with the search results.
